javalib/src/main/scala/java/util/SplittableRandom.scala

```diff
@@ -0,0 +1,136 @@
+package java.util
+
+/*
+ * This is a clean room implementation derived from the original paper
+ * and Java implementation mentioned there:
+ *
+ * Fast Splittable Pseudorandom Number Generators
+ * by Guy L. Steele Jr., Doug Lea, Christine H. Flood
+ * http://gee.cs.oswego.edu/dl/papers/oopsla14.pdf
+ *
+ */
+private object SplittableRandom {
+
+  private final val DOUBLE_ULP = 1.0 / (1L << 53)
+  private final val GOLDEN_GAMMA = 0x9e3779b97f4a7c15L // odd
+
+  //switched from original comparing result of JVM std lib
+  private final def mix64variant13(_z: Long): Long = {
+  //def mix64(_z: Long): Long = {
+    var z = _z
+    z = (z ^ (z >>> 33)) * 0xff51afd7ed558ccdL
+    z = (z ^ (z >>> 33)) * 0xc4ceb9fe1a85ec53L
+    z ^ (z >>> 33)
+  }
+
+  // changed from original trying and comparing result on the JVM for all the numbers mentioned here:
+  // http://zimbry.blogspot.pt/2011/09/better-bit-mixing-improving-on.html
+  // reference was:
+  //def mix32(_z: Long): Int = {
+  //  var z = _z
+  //  z = (z ^ (z >>> 33)) * 0xff51afd7ed558ccdL
+  //  z = (z ^ (z >>> 33)) * 0xc4ceb9fe1a85ec53L
+  //  (z >>> 32).toInt
+  //}
+  // constant here comes from Mix04
```

Comment:
For multi-line comments, please use `/* */`, as shown here: https://github.com/scala-js/scala-js/blob/master/CODINGSTYLE.md#non-scaladoc-comments

I don't think it's useful to have the complete reference in comments here. Instead, say that this function implements David Stafford's variant 4, while the paper version uses the original MurmurHash3 finalizer. The name can stay `mix32` IMO.